lib/os.js

```diff
@@ -121,6 +121,43 @@ function endianness() {
 }
 endianness[Symbol.toPrimitive] = () => kEndianness;
 
+function networkInterfaces() {
+  const interfaceAddresses = getInterfaceAddresses();
+
+  return Object.entries(interfaceAddresses).reduce((acc, [key, val]) => {
+    acc[key] = val.map((v) => {
+      const protocol = v.family.toLowerCase();
+
+      return Object.assign({}, v, {
+        cidr: `${v.address}/${getCIDRSuffix(v.netmask, protocol)}`
+      });
+    });
+    return acc;
+  }, {});
+}
+
+function getCIDRSuffix(ip, protocol = 'ipv4') {
+  let temp;
+
+  if (protocol === 'ipv6') {
+    temp = ip
+      .split(':')
+      .filter((v) => !!v)
+      .map((v) => parseInt(v, 16).toString(2));
+
+  } else {
+    temp = ip
+      .split('.')
+      .map((v) => parseInt(v, 10).toString(2));
+  }
+
+  return temp
+    .join('')
+    .split('')
+    .map((v) => parseInt(v, 10))
+    .reduce((acc, v) => acc += v, 0);
+}
+
```

Comment:
Right, here's a slight modification I came up with that includes validation for invalid masks:

````js
function getCIDRSuffix(subnetMask, proto) {
  const v6 = proto === 'ipv6';
  const parts = subnetMask
    .split(v6 ? ':' : '.')
    .map(p => parseInt(p || 0, v6 ? 16 : 10));

  const isInvalidSubnetMask = parts.some(function(_, i) {
    if (i > 1 && parts[i - 1] < parts[i]) {
      return true;
    }
  });

  if (isInvalidSubnetMask) {
    return null;
  }

  return parts
    .map((dec) => dec.toString(2))
    .join('')
    .split('')
    .map((bin) => parseInt(bin, 10))
    .reduce((acc, v) => acc += v, 0);
}
````